pkg/status/status.go

```diff
@@ -0,0 +1,126 @@
+package status
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"text/template"
+
+	"github.com/stripe/stripe-cli/pkg/ansi"
+)
+
+// Response contains the structure of system statuses from Stripe
+type Response struct {
+	Statuses    statuses `json:"statuses"`
+	LargeStatus string   `json:"largestatus"`
+	Message     string   `json:"message"`
+	Time        string   `json:"time"`
+}
+
+type statuses struct {
+	API        string `json:"api"`
+	Dashboard  string `json:"dashboard"`
+	Stripejs   string `json:"stripejs"`
+	Checkoutjs string `json:"checkoutjs"`
+	Webhooks   string `json:"webhooks"`
+	Emails     string `json:"emails"`
+}
+
+// GetStatus makes a request to the Stripe status site and returns all the
+// current system statuses
+func GetStatus() (Response, error) {
+	var status Response
+
+	resp, err := http.Get("https://status.stripe.com/current")
+	if err != nil {
+		return status, err
+	}
+
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return status, err
+	}
+
+	json.Unmarshal(respBytes, &status)
+	return status, nil
+}
+
+func (r *Response) getMap(verbose bool) map[string]interface{} {
+	responseObject := map[string]interface{}{
+		"status":  r.LargeStatus,
+		"message": r.Message,
+		"time":    r.Time,
+	}
+
+	if verbose {
+		responseObject["statuses"] = map[string]string{
+			"api":        r.Statuses.API,
+			"dashboard":  r.Statuses.Dashboard,
+			"stripejs":   r.Statuses.Stripejs,
+			"checkoutjs": r.Statuses.Checkoutjs,
+			"webhooks":   r.Statuses.Webhooks,
+		}
+	}
+
+	return responseObject
+}
+
+// FormattedMessage returns a properly structured API status response
+// in either a json structure or a templated plain text output, conditionally
+// populated with extra data depending on verbosity
+func (r *Response) FormattedMessage(format string, verbose bool) (string, error) {
+	statusData := r.getMap(verbose)
+	if format == "json" {
+		data, err := json.MarshalIndent(statusData, "", "  ")
+		if err != nil {
+			return "", err
+		}
+
+		return string(data), nil
+	}
+
+	statusString := fmt.Sprintf(`%s %s{{if .statuses}}
+%s API
+%s Dashboard
+%s StripeJS
```

Comment:
Note a big deal, but I wonder if we should use the official product name/case/punctuation here, which I think is "Stripe.js" for StripeJS and just "Checkout" for CheckoutJS.

Reply:
Ah, good call! I'll change this.